src/coreclr/src/vm/reflectioninvocation.cpp

```diff
@@ -2397,7 +2397,7 @@ FCIMPL1(Object *, ReflectionEnum::InternalGetEnumUnderlyingType, ReflectClassBas
     VALIDATEOBJECT(target);
     TypeHandle th = target->GetType();
     if (!th.IsEnum())
-        FCThrowArgument(NULL, NULL);
+        FCThrowArgument(W("enumType"), NULL);
```

Comment:
The exception is still going to have misleading message with this change. You can move this check after `HELPER_METHOD_FRAME_BEGIN_RET_0` and use `COMPlusThrow` that lets you specify the message, like this:

```
    if (!th.IsEnum())
        COMPlusThrow(kArgumentException, W("Arg_MustBeEnum"));
```

Reply:
Sorry if it looks a very basic question, I'm not very expert. But, where the ParamName ("enumType") will be specified with the use of `COMPlusThrow`?

Reply:
I have edited the suggestion